components/prism-liquid.js

```diff
@@ -1,12 +1,35 @@
 Prism.languages.liquid = {
-	'keyword': /\b(?:comment|endcomment|if|elsif|else|endif|unless|endunless|for|endfor|case|endcase|when|in|break|assign|continue|limit|offset|range|reversed|raw|endraw|capture|endcapture|tablerow|endtablerow)\b/,
+	'delimiter': {
+		pattern: /^\{(?:\{\{|[%\{])-?|-?(?:\}\}|[%\}])\}$/,
+		alias: 'punctuation'
+	},
+	'string': {
+		pattern: /"[^"]*"|'[^']*'/,
+		greedy: true
+	},
+	'keyword': /\b(?:as|assign|break|capture|case|comment|contains|continue|decrement|echo|else|elsif|endcapture|endcase|endcomment|endfor|endif|endraw|endtablerow|endunless|for|if|in|include|increment|limit|liquid|offset|range|raw|render|reversed|tablerow|unless|when|with)\b/,
+	'function': {
+		pattern: /(^|[\s;|&])(?:append|prepend|capitalize|cycle|cols|abs|at_least|at_most|ceil|compact|concat|date|default|divided_by|downcase|escape|escape_once|first|floor|join|last|lstrip|map|minus|modulo|newline_to_br|plus|remove|remove_first|replace|replace_first|reverse|round|rstrip|size|slice|sort|sort_natural|split|strip|strip_html|strip_newlines|times|truncate|truncatewords|uniq|upcase|url_decode|url_encode|paginate)(?=$|[\s;:|&])/,
```

Comment:
I would love to be able to get all the function names from the `liquid` gem somehow..

Reply:
Is there a way to detect functions by syntax?

Reply:
I figured it out.